src/alexandria3k/zip_uspto_cache.py

```diff
@@ -0,0 +1,70 @@
+#
+# Alexandria3k Patent grant bibliographic metadata processing
+# Copyright (C) 2023  Aggelos Margkas
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""Cache of decompressed/extracted path of US patent data"""
+
+import zipfile
+
+# Delimiter for extracting concatenated XML files.
+XML_DELIMITER = '<?xml version="1.0" encoding="UTF-8"?>'
+
+
+class ZipCache:
+    """Cache the reading/decompression/extraction of Zip file"""
+
+    # pylint: disable=too-few-public-methods
+    file_reads = 0
+
+    def __init__(self):
+        self.cached_path = None
+        self.cached_data = None
+        self.file_name = None
+
+    def read(self, zip_path):
+        """Takes care of unzipping every Zip file
+        of the US patent dataset once, return
+        a list with all the XML file inside."""
+
+        if zip_path == self.cached_path:
+            return self.cached_data
+
+        with zipfile.ZipFile(zip_path, "r") as zip_ref:
+            # There is only one XML file inside the Zip file.
+            xml_file = [
+                file for file in zip_ref.namelist() if file.endswith(".xml")
+            ]
+
+            # Extracting filename and decoding the XML
```

Comment:
Please write comments in imperative style, i.e. "Extract filename and decode the XML" (look also for other instances).